taf/updater/updater_pipeline.py

```diff
@@ -402,86 +402,101 @@ def set_last_validated_commit(self, validation_repo):
-                        else:
-                            raise UnpushedCommitsError(
-                                auth_repo.name, auth_repo.default_branch
-                            )
+                    repository_not_clean_errors.append(repository.name)
 
-                # check target repositories which are on disk
-                for repository in self.state.repos_on_disk.values():
-                    if repository.is_bare_repository:
+            target = auth_repo.get_target(repository.name)
+            if target and "branch" in target:
+                branch = target["branch"]
+                if repository.is_branch_with_unpushed_commits(branch):
+                    if self.force:
                         taf_logger.info(
-                            f"Skipping clean check for bare repository {repository.name}"
+                            f"Resetting repository {repository.name} to clean state for a forced update."
                         )
+                        repository.clean_and_reset()
                     else:
-                        if repository.something_to_commit():
-                            if self.force:
-                                taf_logger.info(
-                                    f"Resetting repository {auth_repo.name} to clean state for a forced update."
-                                )
-                                repository.clean_and_reset()
-                            else:
-                                raise RepositoryNotCleanError(repository.name)
-
-                        # read the branch from the most recent target files (before the update)
-                        # and check if it contains unpushed commits
-                        # after the update, check if there are unpushed commits on any of the
-                        # other branches
-                        target = auth_repo.get_target(repository.name)
-                        if not target or "branch" not in target:
-                            continue
-                        branch = target["branch"]
-                        if repository.is_branch_with_unpushed_commits(branch):
-                            if self.force:
-                                taf_logger.info(
-                                    f"Resetting repository {auth_repo.name} to clean state for a forced update."
-                                )
-                                repository.clean_and_reset()
-                            else:
-                                raise UnpushedCommitsError(repository.name, branch)
-            return UpdateStatus.SUCCESS
-        except Exception as e:
-            self.state.errors.append(e)
-            self.state.event = Event.FAILED
-            return UpdateStatus.FAILED
+                        unpushed_commits_errors.append((repository.name, branch))
+
+        if repository_not_clean_errors:
+            raise RepositoryNotCleanError(repository_not_clean_errors)
```

Comment:
I think that we probably should combine target and auth repo errors too

Reply:
I've aggregated auth_repo and target repo errors with `MultipleRepositoriesNotCleanError`. How does it look now?